tests/unit/dynamodb/test_transform.py

```diff
@@ -0,0 +1,390 @@
+                'SampleOperation': {
+                    'name': 'SampleOperation',
+                    'input': {'shape': 'SampleOperationInputOutput'},
+                    'output': {'shape': 'SampleOperationInputOutput'}
+                }
+            },
+            'shapes': {
+                self.target_shape_name: {
+                    'type': 'string'
+                },
+                'List': {
+                    'type': 'list',
+                    'member': {'shape': self.target_shape_name}
+                },
+                'Map': {
+                    'type': 'map',
+                    'key': {'shape': 'Name'},
+                    'value': {'shape': self.target_shape_name}
+                },
+                'Structure': {
+                    'type': 'structure',
+                    'members': {
+                        'Member': {'shape': self.target_shape_name}
+                    }
+                },
+                'SampleOperationInputOutput': {
+                    'type': 'structure',
+                    'members': {
+                        'Structure': {'shape': 'Structure'},
+                        'Map': {'shape': 'Map'},
+                        'List': {'shape': 'List'},
+                    }
+                },
+                'Name': {
+                    'type': 'string'
+                }
+            }
+        }
+
+        # Create a more complicated model to test the ability to recurse
+        # through a structure.
+        self.nested_json_model = copy.deepcopy(self.json_model)
+        shapes = self.nested_json_model['shapes']
+        shapes['SampleOperationInputOutput']['members'] = {
+            'Structure': {'shape': 'WrapperStructure'},
+            'Map': {'shape': 'WrapperMap'},
+            'List': {'shape': 'WrapperList'}
+        }
+        shapes['WrapperStructure'] = {
+            'type': 'structure',
+            'members': {'Structure': {'shape': 'Structure'}}
+        }
+        shapes['WrapperMap'] = {
+            'type': 'map',
+            'key': {'shape': 'Name'},
+            'value': {'shape': 'Map'}
+        }
+        shapes['WrapperList'] = {
+            'type': 'list',
+            'member': {'shape': 'List'}
+        }
+
+    def build_models(self):
+        self.service_model = ServiceModel(self.json_model)
+        self.operation_model = OperationModel(
+            self.json_model['operations']['SampleOperation'],
+            self.service_model
+        )
+
+        self.nested_service_model = ServiceModel(self.nested_json_model)
+        self.nested_operation_model = OperationModel(
+            self.nested_json_model['operations']['SampleOperation'],
+            self.nested_service_model
+        )
+
+
+class TestInputOutputTransformer(BaseTransformationTest):
+    def setUp(self):
+        super(TestInputOutputTransformer, self).setUp()
+        self.transformation = lambda params: self.transformed_value
+
+    def test_transform_structure(self):
```

Comment:
Just wanted to capture this discussion here.  It would be great if we could add additional context into this test so that the user doesn't have to scroll around the file to see exactly what's being tested.

I think the test can still be concise, but if could somehow show the linking between the shape name and triggering the transform, that would help with readability:

```
shape_to_transform = 'mytransform'
shape = self.create_input_shape({'Structure': {"TransformMe": shape_to_transform, 'LeaveAlone': 'unmodifiedShape'}})

transformer.transform(shape, ...., target_shape=shape_to_transform)

verify 'TransformMe' was transformed, verify 'LeaveAlone' wasn't.

```

Something like that would make it clear what's going on.